### ISSUE TITLE
Finetune donut model on funsd dataset

### ISSUE THREAD

**Issue by vishal-nayak1** [clovaai/donut#8]
Hi,

Can you please provide finetuning example for **Funsd** dataset as well like data preparation in order to fintune model on funsd dataset?

**Comment by estaudere** [comment]
Hi! Not an author of Donut, but [here's](https://towardsdatascience.com/ocr-free-document-understanding-with-donut-1acfbdf099be) an article I wrote on finetuning Donut on a custom dataset with SROIE. For FUNSD, I've tested that out as well (though it doesn't have a license for commercial use), and you should be able to generate the custom dataset in a similar way I did for SROIE, but where the `gt_parse` object is in the form

```
{
    "questions": [],
    "answers": [],
    "headers: []
}
```

or some other format, if you'd like. Create a new config file in the same way I did for SROIE and you should be able to train with the same script.

**Comment by gwkrsrch** [comment]
Hi, sincerely thanks to @estaudere for the nice introduction!
I read the article and found it very useful.

You may need to explore the proper JSON format.
But I expect following the above introduction will also produce fine results.

**Comment by wdprsto** [comment]
> Hi! Not an author of Donut, but [here's](https://towardsdatascience.com/ocr-free-document-understanding-with-donut-1acfbdf099be) an article I wrote on finetuning Donut on a custom dataset with SROIE. For FUNSD, I've tested that out as well (though it doesn't have a license for commercial use), and you should be able to generate the custom dataset in a similar way I did for SROIE, but where the `gt_parse` object is in the form
> 
> ```
> {
>     "questions": [],
>     "answers": [],
>     "headers: []
> }
> ```
> 
> or some other format, if you'd like. Create a new config file in the same way I did for SROIE and you should be able to train with the same script.

Excuse me, may i know what is the value we should pass inside the "headers" key? And if we have multiple QnAs, does it means we have to write it sequentially inside the list? thank you for your help!

**Comment by samakos** [comment]
@khanfarhan10 yes of course. https://huggingface.co/datasets/SotiriosKastanas/difffunsd here you can find the dataset I prepared. (on in a different format https://huggingface.co/datasets/SotiriosKastanas/Funtry  here but the results are the same. 

And the config I used is exactly the same with CORD config but I changed to 100 epochs.

**Comment by arvisioncode** [comment]
Any update on this?
@samakos Have you found the accuracy issues?

**Comment by Kastanas1** [comment]
I think the model cannot perform well on this task and I am exploring the reasons why
The accuracy I get is 30-35% (F1 score). 
If you have any idea please let me know

**Comment by samakos** [comment]
@arvisioncode I think the model cannot perform well on this task and I am exploring the reasons why
The accuracy I get is 30-35% (F1 score).
If you have any idea please let me know

**Comment by arvisioncode** [comment]
I'm working on this too, but at the moment I don't have good results, although I've tried various approaches to generate the funsd dataset in donut format, they haven't given me good results.
I have also tried doing FT with your data https://huggingface.co/datasets/SotiriosKastanas/diffunsd but I get accuracy 0, after 100 epoch (using test.py)
Obviously, I'm doing something wrong, what base model do you use for the FT? If you have a notebook or something you're working with that you could share with me, it would be a great help!

**Comment by samakos** [comment]
@arvisioncode  okay I will prepare a notebook this weekend and will upload it here

**Comment by arvisioncode** [comment]
hi @samakos 
were you able to prepare this in the end?

**Comment by samakos** [comment]
@arvisioncode  hi, I think I can put here the code because it is only 10 lines!

!git clone https://github.com/clovaai/donut.git
!pip install pytorch-lightning==1.6.4 transformers==4.11.3 timm==0.5.4 sconf zss datasets[vision] sentencepiece
cd donut

!python train.py --config config/train_funsd.yaml \
                --pretrained_model_name_or_path "naver-clova-ix/donut-base" \
                --dataset_name_or_paths '["SotiriosKastanas/difffunsd"]' \
                --exp_version "Experiment_120_train_29valid_200epochs" 

!python test.py --dataset_name_or_path SotiriosKastanas/difffunsd --pretrained_model_name_or_path ./result/train_funsd/Experiment_120_train_29valid_200epochs --save_path ./result/output200_120_29.json


and here is the train_funsd.yaml

resume_from_checkpoint_path: null # only used for resume_from_checkpoint option in PL
result_path: "./result"
pretrained_model_name_or_path: "naver-clova-ix/donut-base" # loading a pre-trained model (from moldehub or path)
dataset_name_or_paths: ["SotiriosKastanas/difffunsd"] # loading datasets (from moldehub or path)
sort_json_key: False # cord dataset is preprocessed, and publicly available at https://huggingface.co/datasets/naver-clova-ix/cord-v2
train_batch_sizes: [2]
val_batch_sizes: [1]
input_size: [1280, 960] # when the input resolution differs from the pre-training setting, some weights will be newly initialized (but the model training would be okay)
max_length: 768
align_long_axis: False
num_nodes: 1
seed: 2022
lr: 3e-5
warmup_steps: 300 # 800/8*30/10, 10%
num_training_samples_per_epoch: 135
max_epochs: 200
max_steps: -1
num_workers: 8
val_check_interval: 1.0
check_val_every_n_epoch: 3
gradient_clip_val: 1.0
verbose: True



the result is F1 score 0.28

**Comment by unhingedpanda** [comment]
Any update on this as I'm trying something similar and been having problems on preparing the dataset for DONUT
